dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/datavalueset/DefaultDataValueSetService.java

```diff
@@ -917,7 +935,8 @@ private void saveDataValueUpdate( ImportContext context, ImportCount importCount
         DataValue internalValue, DataValue existingValue )
     {
         AuditType auditType = AuditType.UPDATE;
-        if ( internalValue.isNullValue() || internalValue.isDeleted() )
+        if ( internalValue.isNullValue() || internalValue.isDeleted()
+            || dataValueIsZeroAndInsignificant( dataValue.getValue(), valueContext.getDataElement() ) )
```

Comment:
New logic: if it is a zero and insignificant update treat this as delete as well. 
This also required that we do not skip this entry because it is zero and insignificant during validation phase so that logic got moved to service to take strategy and existing value into account.